src/OpenCensus/Resource/Resource.cs

```diff
@@ -0,0 +1,159 @@
+﻿// <copyright file="Resource.cs" company="OpenCensus Authors">
+// Copyright 2018, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of theLicense at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+namespace OpenCensus.Resource
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Text.RegularExpressions;
+    using OpenCensus.Tags;
+    using OpenCensus.Utils;
+
+    /// <summary>
+    /// Represents a resource that captures identification information about the entities for which signals (stats or traces)
+    /// are reported. It further provides a framework for detection of resource information from the environment and progressive
+    /// population as signals propagate from the core instrumentation library to a backend's exporter.
+    /// </summary>
+    public abstract class Resource : IResource
```

Comment:
so abstract method is defined. Is there actual implementation or it will be implemented next?

Reply:
It will be implemented next. I wanted to start off with PR to make it easier to review.